analyzer/codechecker_analyzer/cmd/fixit.py

```diff
@@ -0,0 +1,299 @@
+# -------------------------------------------------------------------------
+#
+#  Part of the CodeChecker project, under the Apache License v2.0 with
+#  LLVM Exceptions. See LICENSE for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# -------------------------------------------------------------------------
+"""
+Defines the CodeChecker action for applying fixits.
+"""
+
+import argparse
+import json
+import os
+import re
+import subprocess
+import sys
+import tempfile
+import yaml
+
+from codechecker_analyzer import analyzer_context
+from codechecker_common import arg, logger
+
+LOG = logger.get_logger('system')
+
+
+def get_argparser_ctor_args():
+    """
+    This method returns a dict containing the kwargs for constructing an
+    argparse.ArgumentParser (either directly or as a subparser).
+    """
+
+    return {
+        'prog': 'CodeChecker fixit',
+        'formatter_class': arg.RawDescriptionDefaultHelpFormatter,
+        'description': """
+Some analyzers may suggest some automatic bugfixes. Most of the times these are
+style issues which can be fixed easily. This command handles the listing and
+application of these automatic fixes.
+
+Besides the provided filter options you can pipe the JSON format output of
+"CodeChecker cmd diff" command to apply automatic fixes only for new reports:
+CodeChecker cmd diff -b dir1 -n dir2 -o json --new | CodeChecker fixit dir2""",
+        'help': "Apply automatic fixes based on the suggestions of the "
+                "analyzers"
```

Comment:
Maybe it would be worth mentioning that with the `-i` flag applying the fixits can be done interactively otherwise they are automatic.

Reply:
`-i` flag itself doesn't apply fixits, but it accomplishes an interactive filtering.